esmvaltool/cmorizers/data/formatters/datasets/iap.py

```diff
@@ -0,0 +1,193 @@
+"""
+
+import logging
+import os
+import warnings
+from warnings import catch_warnings
+from datetime import datetime
+from dateutil import relativedelta
+
+import iris
+import cf_units
+import numpy as np
+
+from esmvaltool.cmorizers.data.utilities import (
+    fix_coords,
+    fix_var_metadata,
+    save_variable,
+    set_global_atts,
+)
+
+logger = logging.getLogger(__name__)
+
+try:
+    iris.FUTURE.date_microseconds = True
+    iris.FUTURE.save_split_attrs = True
+except AttributeError as e:
+    # Handle cases where FUTURE or the attributes don't exist
+    logger.warning("AttributeError: %s", e)
+except (TypeError, ValueError) as e:
+    # Handle specific errors if these might occur
+    logger.warning("TypeError or ValueError: %s", e)
+except BaseException as e:
+    # Fallback for rare or unknown issues, but avoid catching Exception
+    logger.warning("An unexpected error occurred: %s", e)
+
+def collect_files(in_dir, cfg, start_date, end_date):
+    """ Create list of files path to be processed."""
+    file_list = []
+
+    if start_date is None:
+        start_date = datetime(year=1940, month=1, day=1)
+    if end_date is None:
+        end_date = datetime(year=2024, month=12, day=31)
+
+    loop_date = start_date
+
+    while loop_date <= end_date:
+        fname = (
+            f"IAPv4_Temp_monthly_1_6000m_year_{loop_date.year}"
+            f"_month_{loop_date.month:02d}.nc"
+        )
+        in_file = os.path.join(in_dir, fname)
+        file_list.append(in_file)
+        loop_date += relativedelta.relativedelta(months=1)
+
+    return file_list
+
+
+def process_data(cube, reference_year):
+    """ Process raw data. Convert to Kelvin and add time dimension.
+        Concatenate the cubes and return the new cube.
+    """
+    # Convert temperature from Celsius to Kelvin and add time dimension
+    temperature_data = np.expand_dims(cube.data, axis=0)
+    temperature_data = np.moveaxis(
+        temperature_data, (0, 1, 2, 3), (0, 2, 3, 1)
+    )  # Reorder axes
+
+    # Create time coordinate
+    start_date = datetime(
+        int(cube.attributes["StartYear"]),
+        int(cube.attributes["StartMonth"]),
+        int(cube.attributes["StartDay"]),
+    )
+    reference_date = datetime(2000, 1, 1)
+    time_points = [(start_date - reference_date).days]
+
+    time_coord = iris.coords.DimCoord(
+        time_points,
+        standard_name="time",
+        units=(
+            f"days since {reference_date.year}-"
+            f"{reference_date.month}-{reference_date.day}"
+        ),
+    )
+
+    # Remove old date attributes
+    for key in ["StartDay", "StartMonth", "StartYear",
+                "EndDay", "EndMonth", "EndYear"]:
+        del cube.attributes[key]
+
+    # Get existing coordinates and rename 'standard depth' to 'depth'
+    latitude_coord = cube.coord("latitude")
+    longitude_coord = cube.coord("longitude")
+    depth_coord = cube.coord("standard depth")
+    depth_coord.rename("depth")
```

Comment:
```suggestion
    depth_coord.rename("depth")
    depth_coord.var_name = "lev"
    depth_coord.attributes['positive'] = "down"
```
I was getting CMOR errors with the depth coordinate var_name, and couldn't load, have added corrections here